proof/crefine/ARM/Retype_C.thy

```diff
@@ -5346,6 +5346,10 @@ lemma pspace_no_overlap_induce_notification:
             split: arch_kernel_object.split_asm)
   done
 
+lemma add_abc_bac:
+  "a + b + c = b + a + (c::'a::semiring)"
+  by (simp add: field_simps)
+
```

Comment:
We should probably leave a FIXME here and clean up the word proof that follows below where this is used. Surely we have better tech and/or insight since the original crimes on that were committed.

(But Ok to leave in for arch split purposes, it's not substantially worse than before)

Reply:
The proof uses extremely specific rewrites, where any simplification or mis-subst stops it from working, hence the extremely specific rewrite to fix it up. It would be interesting to see how much it can be cleaned up. I'll add a normal FIXME for this one then, as it's not very arch-split-specific.

Reply:
The other "solution" is to dump this very specific lemma into Lib, in case anyone ever needs that reordering again.